sklearn/cluster/_birch.py

```diff
@@ -674,8 +674,14 @@ def predict(self, X):
         labels : ndarray of shape(n_samples,)
             Labelled data.
         """
+        return self._predict(X)
+
+    def _predict(self, X, check_input=True):
+        """Predict data using the ``centroids_`` of subclusters."""
         check_is_fitted(self)
-        X = self._validate_data(X, accept_sparse="csr", reset=False)
+        if check_input:
+            X = self._validate_data(X, accept_sparse="csr", reset=False)
+
```

Comment:
I think I prefer having the private function do no validation and do validation + call private function in the public function instead of introducing a `check_input` arg but feel free to ignore if I'm the only one :)